workspaces/ui-v2/src/hooks/useFetchEndpoints.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect } from 'react';
+
+import { useSpectacleContext } from '<src>/contexts/spectacle-provider';
+import { endpointActions, useAppDispatch } from '<src>/store';
+
+export const useFetchEndpoints = () => {
+  const dispatch = useAppDispatch();
+  const spectacle = useSpectacleContext();
+  useEffect(() => {
+    // the typing is correct, we just arent setting it correctly above
+    // TODO remove ts-ignore
+    // @ts-ignore
+    dispatch(endpointActions.fetchEndpoints({ spectacle }));
+  }, [dispatch, spectacle]);
```

Comment:
Also note, this _should_ cause this action to be refetched if the spectacle instance changes (which is what we want). There are two options we have to handle the old data
- send a reload actions that all redux stores reset to loading
- just let all fetch.fulfilled actions overwrite the old store (we need to make sure we build the store with this requirement in mind)